(FOR CI) console: allow using arrow left/right to edit the line

ngxson opened the pull request:
…pport)

*Make sure to read the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md) before submitting a PR*


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Enhanced console navigation: Home/End, Delete, arrow keys with Ctrl‑arrow word movement, consistent behavior across Windows and POSIX.
  * Persistent command history with per-entry backup, Up/Down navigation, and deduplication of saved entries.

* **Bug Fixes**
  * More reliable UTF‑8 decoding and character-aware editing.
  * Improved mid-line edit/redraw and unified escape-sequence handling for consistent cursor positioning.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ngxson committed:
console: allow using arrow left/right to edit the line (with UTF-8 support)

ngxson committed:
console: allow using arrow left/right to edit the line (with UTF-8 support)

ServeurpersoCom committed:
console: fix arrow keys on Windows using private-use Unicode

ServeurpersoCom committed:
console: fix arrow keys on Windows using private-use Unicode

ServeurpersoCom committed:
console: add Home/End key support for Windows and Linux

ServeurpersoCom committed:
console: add Home/End key support for Windows and Linux

ServeurpersoCom committed:
console: add basic Up/Down history navigation

ServeurpersoCom committed:
console: add basic Up/Down history navigation

ngxson committed:
Merge pull request #53 from ServeurpersoCom/xsn/cli_arrow_left_right

Xsn/cli arrow left right

ngxson committed:
fix build

ServeurpersoCom committed:
console: remove unreachable wc == 0 check after VK switch

ServeurpersoCom commented on common/console.cpp:
```diff
@@ -175,6 +186,15 @@ namespace console {
+                }
                 if (wc == 0) {
                     continue;
                 }
```
console: remove unreachable wc == 0 check after VK switch

ServeurpersoCom committed:
console: add Ctrl+Left/Right word navigation

- Add KEY_CTRL_ARROW_LEFT and KEY_CTRL_ARROW_RIGHT codes
- Windows: detect CTRL modifier via dwControlKeyState
- Linux: parse ANSI sequences with modifier (1;5D/C)
- Implement move_word_left/right with space-skipping logic
- Refactor escape sequence parsing to accumulate params

ServeurpersoCom committed:
console: add Delete key support

- Windows: VK_DELETE detection
- Linux: ESC[3~ sequence parsing
- Forward character deletion with UTF-8 support

ServeurpersoCom committed:
console: implement bash-style history editing

- Edit any history line during UP/DOWN navigation, edits persist
- Pressing Enter appends edited version as new history entry
- Original line stay untouched in their positions

ngxson committed:
Merge remote-tracking branch 'pascal/console/terminal-improvement' into xsn/cli_arrow_left_right

ngxson committed:
clean up

ngxson committed:
better history impl

ngxson committed:
fix decode_utf8
